scripts/sentiment_analysis/sentiment_analysis_cnn.py

```diff
@@ -0,0 +1,201 @@
+"""
+TextCNN Model for Sentiment Analysis
+===============================================
+This example shows how to use convolutional neural networks (textCNN)
+for sentiment analysis on various datasets.
+
+Kim, Y. (2014). Convolutional neural networks for sentence classification.
+arXiv preprint arXiv:1408.5882.
+"""
+
+# coding: utf-8
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import time
+import random
+import numpy as np
+
+import mxnet as mx
+from mxnet import nd, gluon, autograd
+from mxnet.gluon.data import DataLoader
+
+import process_data
+import text_cnn
+
+
+np.random.seed(3435)
+random.seed(3435)
+mx.random.seed(3435)
+
+parser = argparse.ArgumentParser(description='Sentiment analysis with the textCNN model on\
+                                 various datasets.')
+parser.add_argument('--data_name', choices=['MR', 'SST-1', 'SST-2', 'Subj', 'TREC'], default='MR',
```

Comment:
This initialy is built for generic text classification usecase,wouldnt it be nice to take in any generic dataset as input and train a classifier from it ? rather than taking only predefined datasets ?

Reply:
Thanks. Due to the nature of scripts (rather than API), we usually dedicate scripts to certain tasks. Users can subclass `SimpleDataset` or tweak the scripts for other tasks. We may extend scripts to allow user-defined data sets as future work.

Reply:
Regarding this request, once the scripts are in good quality, we may start vending it as part of the model zoo API as module functions (e.g. `python -m gluonnlp.model_zoo.sentiment_analysis --input-file 'imdb.txt'`). This is too large a scope to include in 0.5.0. Also, it will be a lot easier to push this forward once we standardize the scripts.

Reply:
Thanks for clarification Sheng!